src/App.js

```diff
@@ -1,37 +1,32 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import * as appActions from './redux/actions';
 import logo from './logo.svg';
 import './App.css';
 import Button from './button';
 import Display from './display';
 import { BUTTON_PROPERTIES_ARRAY } from './constants';
-import { updateResult } from './pressButtonLogic';
 import './calculatorStyle.css';
+import { Provider } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import PropTypes from 'prop-types';
 
 class App extends React.Component {
     constructor(props) {
         super(props);
-        this.state = {
-            result: ['0'],
-            operators: {
-                array: [],
-                redundant: 0,
-            },
-            default: true,
-            point: false,
-        };
-        this.updateState = this.updateState.bind(this);
+        this.handleUpdateState = this.handleUpdateState.bind(this);
     }
 
-    updateState(value) {
-        this.setState({ ...updateResult(value, this.state) });
+    handleUpdateState(value) {
+        this.props.actions.getType(value);
     }
 
     getButtonStyle(properties, index) {
         let newProperties = { ...BUTTON_PROPERTIES_ARRAY[index] };
         if (index === 0) {
```

Comment:
A better approach here is to check first `if (index !== 0) { return newProperties; }` and then handle the case when the index is zero. This way you get rid of one level of indentation.